### PR TITLE
chore: update linter config

### PR DESCRIPTION
**Pull request by xcoulon**
disable 'golint', use 'revive' instead, and apply fixes

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Review by MatousJobanek — APPROVED** [review]

File: .golangci.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@ linters:
   enable:
     - gofmt
     - unparam
+    - revive
```
Comment (MatousJobanek) [inline .golangci.yml:12]: why not `typecheck`?

Comment (xcoulon) [reply .golangci.yml:12]: actually, I'm currently using a more exhaustive list of linters:
```
linters:
  enable:
    - megacheck
    - govet
    - gocyclo
    - unused
    - gofmt
    - revive
    - misspell
    - exportloopref
  presets: # groups of linters. See https://golangci-lint.run/usage/linters/
    - bugs
    - unused
  disable: 
    - golint # deprecated, use 'revive'
    - scopelint # deprecated, use 'exportloopref'
 ```
and `typecheck` is part of the `bugs` preset. See https://golangci-lint.run/usage/linters/

Comment (xcoulon) [reply .golangci.yml:12]: and I think that I can even remove linters already included in the `presets` from the `enabled` list

**Review by ranakan19 — APPROVED** [review]
LGTM
TIL about the nolint directive and presets. Thanks!

**Pull request merged**